### ISSUE TITLE
CSV to XML: Removing spaces in header

### ISSUE THREAD

**Issue by hjoy3005** [Cinchoo/ChoETL#87]
Using ChoCSVReader to read a CSV file and then ChoXmlWriter and I cannot see a way to remove(or replace the spaces in the CSV header names. This causes the XML writer to produce a blank output 

                using (var reader = new ChoCSVReader("C:\\Server Media\\test3.csv").WithFirstLineHeader().Configure(c => c.FileHeaderConfiguration.IgnoreColumnsWithEmptyHeader = true))
                {
                    
                    using (var writer = new ChoXmlWriter(sb).Configure(c => c.RootName = "Records").Configure(c => c.NodeName = "Record").Configure(c => c.EmptyXmlNodeValueHandling= ChoEmptyXmlNodeValueHandling.Empty))
                    {

                        writer.Write(reader);
                    }

                }

in this case it simply returns "<Records></Records>" 

If i manually remove the spaces the the csv file, i.e change "Company Name" to "CompanyName" it works fine.

Am I missing something obvious?

**Comment by Cinchoo** [comment]
Yes, spaces in the header causing xml serialization to fail. I'm planning to put implicit conversion of names on next version. 

In the mean time, you can address it as below

```
            using (var reader = new ChoCSVReader("C:\\Server Media\\test3.csv")
                .WithFirstLineHeader()
                .Configure(c => c.FileHeaderConfiguration.IgnoreColumnsWithEmptyHeader = true)
                )
            {
                using (var writer = new ChoXmlWriter(sb)
                    .Configure(c => c.RootName = "Records")
                    .Configure(c => c.NodeName = "Record")
                    .Configure(c => c.EmptyXmlNodeValueHandling = ChoEmptyXmlNodeValueHandling.Empty)
                    .Configure(c => c.ErrorMode = ChoErrorMode.ThrowAndStop)
                    )
                {
                    writer.Write(reader.Select(r =>
                    {
                        r.RenameKey("Company Name", "CompanyName");
                        return r;
                    }));
                }
            }
```

**Comment by Cinchoo** [comment]
On the next version, it will be handled implicitly by the library.

**Comment by hjoy3005** [comment]
Thanks for the response for now I have handled it as below

                    writer.Write(reader.Select(r =>
                    {
                        foreach (var key in r.KeysArray)
                        {
                            string keyString = key;
                            string newKey = keyString.Replace(" ", string.Empty);
                            r.RenameKey(keyString, newKey);
                        }
                        return r;
                    }));

**Comment by Cinchoo** [comment]
Well, that's good. 

new package v1.2.0.2 released to handle it. pls try and let me know. thx

**Comment by hjoy3005** [comment]
Thanks, cant seem to make it work( I just used the code as per the OP)

**Comment by hjoy3005** [comment]
I have been playing around with this and it seems to work for some scenarios.

I have noticed some strangeness 
in my CSV I have 2 column headers "Buyer Fax" and "Buyer Contact"

I can see how this is handled in the "Alternate keys"

![image](https://user-images.githubusercontent.com/65598494/83027623-39f95900-a028-11ea-99ce-99edbeb50132.png)


I have manaully renamed the keys myself using underscores to 

![image](https://user-images.githubusercontent.com/65598494/83027669-48477500-a028-11ea-90a8-877ba5bda9f9.png)


however in the resulting XML it seems to use the alternate key for one and the manually renmaed key for the other 

![image](https://user-images.githubusercontent.com/65598494/83027747-6614da00-a028-11ea-94fb-ca57a9f485a6.png)


howver if there is a value (in this case BuyerFax) it uses the Alternate key 
![image](https://user-images.githubusercontent.com/65598494/83028024-c99f0780-a028-11ea-94eb-e92b33805c71.png)

It seems to use the Alternate Key if there is a value, and the  key of there is not
